src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -528,27 +528,31 @@ int uct_rc_mlx5_iface_is_reachable(const uct_iface_h tl_iface,
 }
 
 
-static int uct_rc_mlx5_iface_srq_topo(uct_rc_mlx5_iface_common_t *iface,
-                                      uct_rc_mlx5_iface_common_config_t *mlx5_config,
-                                      int dc)
+static ucs_status_t
+uct_rc_mlx5_iface_srq_topo(uct_rc_mlx5_iface_common_t *iface,
+                           uct_rc_mlx5_iface_common_config_t *mlx5_config,
+                           int qp_type, int *topo)
 {
-    int srq_cyclic_reqiured  = (mlx5_config->srq_topo == UCT_RC_MLX5_SRQ_TOPO_CYCLIC);
-    int srq_cyclic_supported = uct_rc_mlx5_is_devx_srq_supported(iface, 1);
+    int dc                   = (qp_type == UCT_IB_QPT_DCI);
+    int srq_cyclic_requested = (mlx5_config->srq_topo == UCT_RC_MLX5_SRQ_TOPO_CYCLIC);
+    int srq_cyclic_supported = uct_rc_mlx5_is_devx_srq_supported(iface, dc);
 
-    if (srq_cyclic_reqiured ||
+    if (srq_cyclic_requested ||
         (mlx5_config->srq_topo == UCT_RC_MLX5_SRQ_TOPO_AUTO)) {
-        if (!srq_cyclic_supported && srq_cyclic_reqiured) {
-            ucs_error("Cyclic SRQ WQ type was requested, but it is not "
-                      "supported by the configuration");
-            return UCT_IB_MLX5_SRQ_TOPO_UNKNOWN;
+        if (!srq_cyclic_supported && srq_cyclic_requested) {
+            ucs_error(UCT_IB_IFACE_FMT": cyclic SRQ WQ type is not supported",
+                      UCT_IB_IFACE_ARG(&iface->super.super));
+            return UCS_ERR_UNSUPPORTED;
         }
 
-        return UCT_RC_MLX5_MP_ENABLED(iface) ?
-               UCT_IB_MLX5_SRQ_TOPO_CYCLIC_MP_RQ : UCT_IB_MLX5_SRQ_TOPO_CYCLIC;
+        *topo = UCT_RC_MLX5_MP_ENABLED(iface) ?
+                UCT_IB_MLX5_SRQ_TOPO_CYCLIC_MP_RQ : UCT_IB_MLX5_SRQ_TOPO_CYCLIC;
+        return UCS_OK;
     }
 
-    return UCT_RC_MLX5_MP_ENABLED(iface) ?
-           UCT_IB_MLX5_SRQ_TOPO_LIST_MP_RQ : UCT_IB_MLX5_SRQ_TOPO_LIST;
+    *topo = UCT_RC_MLX5_MP_ENABLED(iface) ?
+            UCT_IB_MLX5_SRQ_TOPO_LIST_MP_RQ : UCT_IB_MLX5_SRQ_TOPO_LIST;
+    return UCS_OK;
```

Comment:
could be done simpler+ remove some local vars:
```
if (config->topo == LIST) {
 *topo = UCT_RC_MLX5_MP_ENABLED(iface) ? 
      UCT_IB_MLX5_SRQ_TOPO_LIST_MP_RQ : UCT_IB_MLX5_SRQ_TOPO_LIST;
  return OK;
 }

srq_cyclic_supported = uct_rc_mlx5_is_devx_srq_supported(iface, qpt==DC);
if (config->topo == CYCLIC && !srq_cyclic_supported) {
   ucs_error
   return UNSUPPORTED;
 }

*topo = UCT_RC_MLX5_MP_ENABLED(iface) ? UCT_IB_MLX5_SRQ_TOPO_CYCLIC_MP_RQ : UCT_IB_MLX5_SRQ_TOPO_CYCLIC;
return OK;
```

Reply:
done